src/utils/generator.js

```diff
@@ -748,6 +748,17 @@ export class Generator {
         };
         this.emitInvariant([value, value], condition, valueNode => valueNode);
       }
+    } else if (value instanceof FunctionValue) {
+      // We do a special case for functions,
+      // as we like to use concrete functions in the model to model abstract behaviors.
+      // These concrete functions do not have the right identity.
+      condition = ([objectNode]) =>
+        t.binaryExpression(
+          "!==",
+          t.unaryExpression("typeof", accessedPropertyOf(objectNode), true),
+          t.stringLiteral("function")
+        );
+      this.emitInvariant([object, value, object], condition, objnode => accessedPropertyOf(objnode));
```

Comment:
This will do nothing if realm.omitInvariants is true. We may as well check this at the start of emitFullInvariant and avoid some work.